internal/pkg/cli/pipeline_init.go

```diff
@@ -0,0 +1,153 @@
+	// 	if err := opts.askGitHubAccessToken(); err != nil {
+	// 		return err
+	// 	}
+	// }
+
+	// if len(opts.Environments) == 0 {
+	// 	if err := opts.askEnvironments(); err != nil {
+	// 		return err
+	// 	}
+	// }
+	return nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (opts *InitPipelineOpts) Validate() error {
+	// TODO
+	return nil
+}
+
+// Execute writes the pipline manifest file.
+func (opts *InitPipelineOpts) Execute() error {
+	// TODO
+	return nil
+}
+
+func (opts *InitPipelineOpts) askAppName() error {
+	// projectName := viper.GetString(projectFlag)
+	// apps, err := opts.appStore.ListApplications(projectName)
+	// if err != nil {
+	// 	return err
+	// }
+
+	// selections := make([]string, len(apps))
+	// for _, app := range apps {
+	// 	selections = append(selections, app.Name)
+	// }
+
+	// sel, err := opts.prompt.SelectOne(
+	// 	fmt.Sprintf("Which application would you like to create a pipeline for?"),
+	// 	`The application to create a pipeline for.`,
+	// 	selections,
+	// )
+	// if err != nil {
+	// 	return fmt.Errorf("failed to select application: %w", err)
+	// }
+
+	return nil
+}
+
+// BuildPipelineInitCmd build the command for creating a new pipeline.
+func BuildPipelineInitCmd() *cobra.Command {
+	opts := &InitPipelineOpts{}
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Creates a pipeline for applications in your workspace.",
+		Long: `Creates a pipeline for the applications in your workspace, using the environments associated with the applications."`,
+		Example: `
+  Create a pipeline for the applications in your workspace
+  /code $ archer pipeline init \
+    --github-repo "gitHubUserName/myFrontendApp" \
+    --github-access-token file://myGitHubToken \
+    --environments "stage,prod" \
+    --deploy`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.fs = &afero.Afero{Fs: afero.NewOsFs()}
+			opts.prompt = prompt.New()
+
+			store, err := ssm.NewStore()
+			if err != nil {
+				return fmt.Errorf("couldn't connect to project datastore: %w", err)
+			}
+			opts.appStore = store
+
+			ws, err := workspace.New()
+			if err != nil {
+				return fmt.Errorf("workspace cannot be created: %w", err)
+			}
+			opts.manifestWriter = ws
+
+			return opts.Validate()
+		},
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Warningln("It's best to run this command in the root of your workspace.")
+			if err := opts.Ask(); err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil { // validate flags
+				return err
+			}
+			return opts.Execute()
+		},
+	}
+	cmd.Flags().StringVarP(&opts.GitHubRepo, "github-repo", "r", "", "GitHub repository for your application.")
+	cmd.Flags().StringVarP(&opts.GitHubAccessToken, "github-access-token", "t", "", "GitHub personal access token for your GitHub repository.")
+	cmd.Flags().BoolVarP(&opts.Deploy, "deploy", "d", false, "Deploy pipline automatically.")
```

Comment:
I can't think of a better description of this - but we should noodle on it a bit. It's like, we're going to actually create the Pipeline after we generate the pipeline.yml for it. Hm - if you can think of something that expresses that in a non-confusing way I'll give you POINTZ